components/brave_wallet/browser/eth_data_parser.cc

```diff
@@ -138,7 +180,7 @@ GetTransactionInfoFromData(const std::vector<uint8_t>& data) {
                                  "uint256"},  // taker amount
         std::vector<std::string>{fill_path,
                                  "",  // maker asset is ETH, amount is txn value
-                                 tx_args.at(1)});
+                                 decoded_calldata.value()[1].GetString()});
```

Comment:
Why these tx_args are not  
```
decoded_calldata.value()[0].GetString(),
decoded_calldata.value()[1].GetString(),
decoded_calldata.value()[2].GetString(),
```
?

Reply:
The `tx_args` corresponds to the `ETHSwap` transaction type. The idea is to parse the calldata using the relevant ABI but eventually format it as per the `tx_args` expected by `ETHSwap`.